Store configuration in non-volatile memory

samuelsadok opened the pull request:
Use the last two flash pages to store configuration data across reboots.

To test:
```
>>> my_odrive.motor0.pos_gain
20.0
>>> my_odrive.motor0.pos_gain = 30
>>> my_odrive.save_config()
[power cycle ODrive]
>>> my_odrive.motor0.pos_gain
30.0
```

 ~~- [ ] check whether the modifications to the linker script are in conflict with the STM code generator~~ this is easier to check on the `sam_cubemx_cleanup` branch once this one is merged

samuelsadok committed:
Add low level non-volatile memory implementation

samuelsadok committed:
Add persistent storage of certain motor parameters.

The persistent storage can be controlled through the "save_config" and "reset" RPC functions. If a valid configuration is present, it's automatically loaded on startup. For now only a few parameters are saved (see config.cpp).

samuelsadok committed:
add reference to datasheet

samuelsadok committed:
add note about firmware update

madcowswe commented on Firmware/MotorControl/config.cpp:
```diff
@@ -0,0 +1,231 @@
+/* Private variables ---------------------------------------------------------*/
+/* Private function prototypes -----------------------------------------------*/
+/* Function implementations --------------------------------------------------*/
+
```
Please comment this machinery.

madcowswe committed:
change how default configs are done for now

madcowswe committed:
add axis to config

samuelsadok committed:
add version number for NVM config, add code comments

samuelsadok committed:
add support for bool to protocol

samuelsadok committed:
Reorganize config parameters

All configurable parameters are now exposed on the protocol in objects called "[...].config". Those are the same parameters that are saved to NVM.

madcowswe committed:
Update README.md

madcowswe committed:
Update README.md

madcowswe committed:
refactor readme

madcowswe committed:
Merge branch 'devel' into sam_nvm

madcowswe committed:
Update Tupfile.lua

madcowswe committed:
fix comma

madcowswe committed:
elec_rad_per_enc also evaluated in check CPR

madcowswe committed:
Merge branch 'devel' into sam_nvm

madcowswe committed:
fix saving of calibrated flag
